Admin UI Issue #5904 - Fix incorrect public links

kcristiano opened the pull request:
Migrate to Civi::url to allow context to be passed from SK Links

Overview
----------------------------------------
See https://lab.civicrm.org/dev/core/-/issues/5904


Before
----------------------------------------
User Facing links all are backend or front end depending on context of the browser.   Any links viewed in `wp-admin` on WP are backend even if it is a public facing donation page.

After
----------------------------------------
Allow URLs to specify frontend or backend.



Technical Details
----------------------------------------
Switch from `CRM_Utils_System::url` to `Civi::url`.  Default existing links that have no scheme (frontend, backend) to use `current` scheme.

An additional change was needed to `Civi/Core/Url.php` to ensure that is an empty string (as opposed to a NULL) was returned for scheme that we would respect the `current` context.

Follow up PR will be done to update the Admin UI extension.


Comments
----------------------------------------
cc @aydun @colemanw @totten

kcristiano committed:
Admin UI Issue #5904 - Migrate to Civi::url to allow context to be passed from SK Links

Only set absolute flag when action is Download

colemanw commented on ext/search_kit/Civi/Api4/Action/SearchDisplay/AbstractRunAction.php:
```diff
@@ -987,7 +987,7 @@ private function getUrl(string $path, $query = NULL) {
     // Use absolute urls when downloading spreadsheet
     $absolute = $this->getActionName() === 'download';
-    return \CRM_Utils_System::url($path, $query, $absolute, NULL, FALSE);
+    return (string) \Civi::url($path, 'a')->addQuery($query);
```
@kcristiano I think you're using the `'a'` flag unconditionally here but it should depend on the value of `$absolute`

colemanw commented on ext/search_kit/Civi/Api4/Action/SearchDisplay/AbstractRunAction.php:
```diff
@@ -986,8 +986,8 @@ private function getUrl(string $path, $query = NULL) {
     // Use absolute urls when downloading spreadsheet
-    $absolute = $this->getActionName() === 'download';
-    return \CRM_Utils_System::url($path, $query, $absolute, NULL, FALSE);
+    $flags = $this->getActionName() === 'download';
+    return (string) \Civi::url($path, $flags ? 'a' : null)->addQuery($query);
```
```suggestion
    $flags = NULL;
    if ($this->getActionName() === 'download') {
      $flags = 'a';
    }
    return (string) \Civi::url($path, $flags)->addQuery($query);
```

kcristiano committed:
Set Links to frontend when link is public

Updates Manage Contribution Pages, PCP Page admin listing, and Administer profiles

aydun commented:
Tested - seems good.
